src/torrent.cpp

```diff
@@ -8873,31 +8863,39 @@ namespace libtorrent
 		announce_with_tracker(tracker_request::stopped);
 	}
 
-	int torrent::finished_time() const
+	seconds_t torrent::finished_time() const
 	{
-		// m_finished_time does not account for the current "session", just the
-		// time before we last started this torrent. To get the current time, we
-		// need to add the time since we started it
-		return m_finished_time + ((!is_finished() || is_paused()) ? 0
-			: (m_ses.session_time() - m_became_finished));
+		if(!is_finished() || is_paused())
+			return m_finished_time;
+		// cast ratio<1l, 1000000000l> to seconds_t ratio
```

Comment:
I think you should remove this comment too. that ratio you specify is implementation defined and not really relevant here.
This applies the the same comments below as well